pkg/secrethub/credentials/creators.go

```diff
@@ -0,0 +1,51 @@
+package credentials
+
+import (
+	awssdk "github.com/aws/aws-sdk-go/aws"
+
+	"github.com/secrethub/secrethub-go/internals/aws"
+)
+
+type Creator interface {
+	Create() (Verifier, Encrypter, error)
+}
+
+type KeyCreator struct {
+	key *RSACredential
+}
+
+func CreateKey() *KeyCreator {
+	return &KeyCreator{}
+}
+
+func (c *KeyCreator) Create() (Verifier, Encrypter, error) {
+	key, err := GenerateCredential()
+	if err != nil {
+		return nil, nil, err
+	}
+	c.key = key
+	return c.key, c.key, nil
+}
+
+func (c *KeyCreator) Export() []byte {
+	if c.key == nil {
+		return nil
```

Comment:
Should this return `nil` or a zero `[]byte{}`?

Reply:
For almost all intents and purposes that is the same. Only difference I know is how they're JSON-encoded. In that case `nil` makes more because it is "not set", the key is not of length 0.

Altenative would be to add an error to this function.